src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -294,6 +296,21 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
       return
     }
 
+    const messageType = mavlink_message.message.type
+
+    // Special handling for NAMED_VALUE_FLOAT messages
+    if (messageType === 'NAMED_VALUE_FLOAT') {
+      const name = (mavlink_message.message.name as string[]).join('').replace(/\0/g, '')
+      setDataLakeVariableData(`${messageType}/${name}`, mavlink_message.message.value)
+      return
+    }
+
+    // For all other messages, use the flattener
+    const flattened = flattenData(mavlink_message.message)
+    flattened.forEach(({ path, value }) => {
+      setDataLakeVariableData(path, value)
+    })
+
```

Comment:
Should this be in the switch case of line 524 (master)?

Reply:
it could. I did steer away from it as I have a feeling we'll move away from the "genericVariables" logic and into the datalake instead. So I kept things separated already.

But I don't mind moving it there, your call

Reply:
Makes sense. Let's stay as it is.